### DISCUSSION TITLE
Cask/Desc: Description shouldn't start with the cask name

### DISCUSSION THREAD

**Discussion by alexreg** [Homebrew/discussions#4109] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.18-11-gdef85c7
ORIGIN: https://github.com/Homebrew/brew
HEAD: def85c7800175b09cf21d2f756319323adc6ac14
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: (none)
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_BROWSER: open -a Firefox.app
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.CP1wa6YEjt/org.xquartz:0
HOMEBREW_EDITOR: mate_wait
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.39.0 => /opt/homebrew/bin/git
Curl: 7.85.0 => /usr/bin/curl
macOS: 13.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  subliminal

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/BandedSmithWaterman.h
  /usr/local/include/BedReader.h
  /usr/local/include/Fasta.h
  /usr/local/include/IndelAllele.h
  /usr/local/include/IntervalTree.h
  /usr/local/include/LargeFileSupport.h
  /usr/local/include/LeftAlign.h
  /usr/local/include/Mosaik.h
  /usr/local/include/Region.h
  /usr/local/include/Repeats.h
  /usr/local/include/SmithWatermanGotoh.h
  /usr/local/include/Variant.h
  /usr/local/include/disorder.h
  /usr/local/include/filevercmp.h
  /usr/local/include/join.h
  /usr/local/include/makeUnique.h
  /usr/local/include/mt19937ar.h
  /usr/local/include/multichoose.h
  /usr/local/include/multipermute.h
  /usr/local/include/vec128int.h
  /usr/local/include/veclib_types.h
```

### Description of issue

The style error `Cask/Desc: Description shouldn't start with the cask name` is problematic in that it reveals false positives sometimes. For example, I have a cask in a private tap called *dedrm*, with the description "DeDRM tools for e-books" (the official description for the project). Clearly, "DeDRM" is being used in the description as a description, *not* the name of the cask. And even if I change it to "De-DRM" (for example), the style error persists. This is clearly undesirable. So, is there any way to suppress this style error?

**Comment by alexreg** [comment]
Hello. Any thoughts on this?

**Reply by gromgit** [reply]
Not a maintainer, but I don't see why it's imperative to stick to the project's GitHub blurb. In fact, the project's original author provided [an alternate description](https://github.com/noDRM/DeDRM_tools#original-readme-from-apprentice-harper) that passes the style check:
> This is a repository that tracks all the scripts and other _**tools for removing DRM from ebooks**_ that I could find, committed in date order as best as I could manage.

**Reply by alexreg** [reply]
It's not, but the description is perfectly reasonable here. This is clearly the case of an over-zealous linter.

**Reply by alebcay** [reply]
My understanding is that this lint rule is usually to prevent descriptions of the form "1Password is a password manager that keeps all passwords secure behind one password" or "Firefox is a web browser".

It's just easier and more reasonable to check for the formula/cask name being the first word than it is to import an NLP framework here to validate the description text.

Also note that `brew desc` already prints the name when it prints the description, this is also probably another reason to avoid repeating the name again:

```
$ brew desc --cask firefox
firefox: (Mozilla Firefox) Web browser
```

**Reply by alexreg** [reply]
@alebcay Oh yes, it's definitely for that reason. I'm just saying it probably makes sense to be able to ignore the lint in cases when it's overzealous, like above.

**Reply by alexreg** [reply]
For the time being, @gromgit's suggestion is a fair one however. Thank you for both your replies.
